datajoint/expression.py

```diff
@@ -187,6 +217,13 @@ def restrict(self, restriction):
         string, or an AndList.
         """
         attributes = set()
+        if isinstance(restriction, Top):
+            self._top = dict(
+                limit=restriction.limit,
+                offset=restriction.offset,
+                order_by=restriction.order_by,
+            )
+            return self.make_subquery()
```

Comment:
Is this the right time to make a subquery? I think the subquery should be created when applying other operators after `dj.Top` has been applied.